### DISCUSSION TITLE
How to disable preprocessed hits?

### DISCUSSION THREAD

**Discussion by zmike** [ccache/ccache#283] [Questions & answers] [answered]
I'm trying to optimize a build, and I have two results on CI:

Fast:
https://travis-ci.org/Enlightenment/efl/jobs/402876913

Slow:
https://travis-ci.org/Enlightenment/efl/jobs/403656274

The slow build takes 2x longer than the fast build despite having the same cache size and the slow build having a much higher cache hit rate. I'd like to completely disable preprocessed cache hits for testing, since it seems like this is likely to be the cause of the slowdown?

Does such a functionality exist?

**Comment by afbjorklund** [comment, marked as answer]
You can make the cache read-only and you can disable direct mode, but currently the preprocessor *needs* to be used in order to fill the cache. Enabling (and storing) the ccache logfile would probably be the first step of investigation.

**Comment by zmike** [comment]
Oh that's interesting, I hadn't looked deeply into the implementation regarding preprocessor being required. Thanks!

I resolved the issue here: it turned out that the build was adding some `#define` (never used by anything) to a header which was different in every build, effectively invalidating the cache.

**Comment by afbjorklund** [comment]
We are looking at an improvement, to use the dependency files instead of the preprocessor: #234
